### PR TITLE
OCaml improvements

### PR DESCRIPTION
**Pull request by emillon**
Hi!

I use this gem as part of gitlab and noticed that it generated some errors in OCaml files. I added some tests, not sure whether it was useful to change examples as well.

Let me know what you think.

Thanks for maintaining this gem! ✨

### CODE REVIEW

File: spec/lexers/ocaml_spec.rb [hunk 1]
```diff
@@ -17,5 +17,31 @@
...
+    end
+
+    it 'recognizes polymorphic variants' do
+      assert_no_errors '`max'
+    end
+
+    it 'recognizes identifiers that start with _' do
+      assert_no_errors 'let x = M._x'
+    end
+  end
```
Comment (jneen) [inline spec/lexers/ocaml_spec.rb]: Could you add these to the visual sample file instead of the spec file? It's in `spec/visual/samples/ocaml`.

**Comment by jneen** [comment]
Thanks! Just a slight change for the tests, but this looks good.

File: spec/lexers/ocaml_spec.rb (hunk 1 above)
Comment (emillon) [inline spec/lexers/ocaml_spec.rb]: done. is there a way to automatically check that those visual samples do not have errors?

**Comment by emillon** [comment]
Just fixed the tests and rebased to solve the conflicts!

**Comment by jneen** [comment]
Thanks!

**Pull request merged**